src/zope/container/contained.py

```diff
@@ -827,9 +824,9 @@ def chooseName(self, name, object):
         # convert to unicode and remove characters that checkName does not allow
         if isinstance(name, bytes):
             name = name.decode()
-        if not isinstance(name, unicode):
+        if not isinstance(name, text_type):
             try:
-                name = unicode(name)
+                name = text_type(name)
             except:
```

Comment:
That dates back to before Python 3 support landed. Since `__unicode__` is a thing (as is `__str__` for Py3) they could theorecitally throw just about anything. I can change that to a `UnicodeError`, but I'm not sure that's correct.